packages/eslint-plugin-query/src/rules/prefer-query-object-syntax/prefer-query-object-syntax.test.ts

```diff
@@ -3,7 +3,14 @@ import { normalizeIndent } from '../../utils/test-utils'
 import { name, rule } from './prefer-query-object-syntax'
 
 const ruleTester = new ESLintUtils.RuleTester({
-  parser: '@typescript-eslint/parser',
+  /** HACK
+   * We are using a newer version of typescript than the one at the repository root,
+   * to be able to test code that uses newer typescript features.
+   * To prevent RuleTester from using the typescript version at the root
+   * we resolve '@typescript-eslint/parser' here instead of letting it do it internally.
+   * Once the version at the root is updated to >=4.9 this can be replaced by just "@typescript-eslint/parser"
+   */
+  parser: require.resolve('@typescript-eslint/parser') as any,
```

Comment:
where are we using an older version of `@typescript-eslint/parser` ?

Reply:
here: https://github.com/TanStack/query/blob/ef5125f68bc41c6c01219f7cb4b098b8d35a2045/package.json#L61

I came to the conclusion that I also needed to update typescript to >=4.9 to be able to test `satisfies` and I didn't wan't to change the typescript version in the respository root. But maybe I can update @typescript-eslint in the root. I will give it another look.

Reply:
I have tried updating only `@typescript-eslint/*` packages to oldest version that suports `satisfies` (5.45.0), but that is not enough, `typescript` >=4.9 is also needed, otherwise the test that uses `satisfies` fails with `A fatal parsing error occurred: Parsing error: ',' expected.`

I didn't want to update the typescript version in the root, because I assumed that it is probably there to ensure that packages stay compatible with that version of typescript.

Reply:
ah yes, I see. what we'd want is to develop with TS 5.2, but have a type-testing suite that works against all sorts of TS versions.

@ZackDeRose started with this some time back, but we never followed through:

- https://github.com/TanStack/query/pull/5640

I'll take a look at setting this up later this week, then we should be good without this workaround 👍